src/memory/paging/mod.rs

```diff
@@ -317,9 +324,18 @@ pub fn remap_the_kernel<FA>(active_table: &mut ActivePageTable,
     let mut new_table = {
         let frame = allocator.allocate_frame()
             .expect("No more frames");
-        InactivePageTable::new(frame, &mut ACTIVE_TABLE.lock(), &mut temporary_page)
+        InactivePageTable::new(frame, &mut ACTIVE_TABLE.lock(), &mut temporary_page, false)
     };
 
+    //FIXME: for some reason the p3 at the 511th position 
+    //       in this original p4 is not recursively mapped correctly.
+    //       This code triggers that.
+    //
+    //serial_println!("About to try to map at 511th p4, 510th p3, 0th p2");
```

Comment:
I don't understand this print statement. there is only one p4 table. Also we recurvisely map at the 510th index, not 511

Reply:
the p3 at 511 shouldn't be recursively mapped

Reply:
If that's true, then that's the answer to the other problems we've been having then. The 511th p3 isn't recursively mapped. When we created a new p4 from scratch, when it mapped the kernel it just created new tables at the 511th position, but when I had it copying in the old p4 entries on the higher half, it tried to map something to that non-recursively-mapped page, causing a problem. The solution is to just not copy the entries over in this situation.